core/types/system_tx.go

```diff
@@ -10,7 +10,8 @@ import (
 
 type SystemTx struct {
 	Sender common.Address  // pre-determined sender
-	To     *common.Address // system contract
+	To     common.Address  // system contract address
+	Nonce  uint64          // nonce of sender account
```

Comment:
Why add nonce? Can't we just bump it implicly as we do for `L1MessageTx`?

Reply:
to make it easier to track the tx ordering. `L1MessageTx` has the L1 msg index to track the ordering but `SystemTx` has none if there's no nonce

Reply:
But the sender (sequencer) has full control over the ordering anyway. And we can add additional checks in the contract itself.

Reply:
removed